api/src/main/java/io/jsonwebtoken/JwtBuilder.java

```diff
@@ -523,14 +525,46 @@ public interface JwtBuilder extends ClaimsMutator<JwtBuilder> {
      *
      * <p>This is a convenience wrapper for:</p>
      * <blockquote><pre>
-     * {@link #claims()}.{@link ClaimsMutator#expiration(Date) expiration(exp)}.{@link BuilderClaims#and() and()}</pre></blockquote>
+     * {@link #claims()}.{@link ClaimsMutator#expiration(Instant) expiration(exp)}.{@link BuilderClaims#and() and()}</pre></blockquote>
      *
      * @param exp the JWT {@code exp} value or {@code null} to remove the property from the Claims map.
      * @return the builder instance for method chaining.
      */
     @Override
     // for better/targeted JavaDoc
-    JwtBuilder expiration(Date exp);
```

Comment:
If adding these other wrappers, keeping the ability to wrap a `Date` would help folks, and lessen the burden when migrating

Reply:
personally I would avoid keeping `java.util.Date` around even as a convenience or temporary warpper

This would require a java version increase, and thus mean a major release. In my opinion it would be the ideal time to 'force' the devs to fix the breaking changes done in the major bump.

ofc, I am not the maintainer and will bend to the will of the code owner. just my 2cents